api/filter_dimension_options.go

```diff
@@ -28,22 +26,39 @@ func (api *FilterAPI) getFilterBlueprintDimensionOptionsHandler(w http.ResponseW
 		"filter_blueprint_id": filterBlueprintID,
 		"dimension":           dimensionName,
 	}
+	offsetParameter := r.URL.Query().Get("offset")
+	limitParameter := r.URL.Query().Get("limit")
 
 	ctx := r.Context()
 	log.Event(ctx, "get filter blueprint dimension options", log.INFO, logData)
 
-	// get limit from query parameters, or default value
-	limit, err := getPositiveIntQueryParameter(r.URL.Query(), "limit", api.defaultLimit)
-	if err != nil {
-		log.Event(ctx, "failed to obtain limit from request query parameters", log.ERROR, logData)
-		setErrorCode(w, err)
-		return
+	offset := api.defaultOffset
+	limit := api.defaultLimit
+	var err error
+
+	if offsetParameter != "" {
+		logData["offset"] = offsetParameter
+		offset, err = validatePositiveInt(offsetParameter)
+		if err != nil {
+			log.Event(ctx, "failed to obtain offset from request query parameters", log.ERROR, logData)
+			setErrorCode(w, err)
+			return
+		}
 	}
 
-	// get offset from query parameters, or default value
-	offset, err := getPositiveIntQueryParameter(r.URL.Query(), "offset", api.defaultOffset)
-	if err != nil {
-		log.Event(ctx, "failed to obtain offset from request query parameters", log.ERROR, logData)
+	if limitParameter != "" {
+		logData["limit"] = limitParameter
+		limit, err = validatePositiveInt(limitParameter)
+		if err != nil {
+			log.Event(ctx, "failed to obtain limit from request query parameters", log.ERROR, logData)
+			setErrorCode(w, err)
+			return
+		}
+	}
+
+	if limit > api.maxLimit {
```

Comment:
Is there value in putting the max limit in the log data? Seems like it would be useful for anyone investigating an issue in this area.

Reply:
good point, I will put it